core/src/actors/codec.rs

```diff
@@ -25,25 +25,34 @@ pub enum Response {
 }
 
 /// Codec for client -> server transport
+///
+/// Format:
+/// ```norun
+/// Message size: u16
+/// Message: [u8; Message size]
+/// ```
+///
+/// The message format is described in the file [schemas/protocol.fbs][protocol]
+///
+/// [protocol]: https://github.com/witnet/witnet-rust/blob/master/schemas/protocol.fbs
 #[derive(Debug, Message, Eq, PartialEq, Clone)]
 pub struct P2PCodec;
 
 /// Implement decoder trait for P2P codec
-///
 impl Decoder for P2PCodec {
     type Item = Request;
     type Error = io::Error;
 
     /// Method to decode bytes to a request
     fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
         let mut ftb: Option<Self::Item> = None;
-        let msg_len = src.len() as u16;
+        let msg_len = src.len();
         if msg_len > 2 {
-            let mut header_vec = Cursor::new([src.to_vec()[0], src.to_vec()[1]]);
-            let msg_size = header_vec.read_u16::<BigEndian>().unwrap();
-            if msg_len >= (msg_size + HEADER_SIZE) {
-                src.split_to(HEADER_SIZE as usize);
-                ftb = Some(Request::Message(src.split_to(msg_size as usize)));
+            let mut header_vec = Cursor::new(&src[0..2]);
```

Comment:
:ok_hand:

Reply:
:heart_eyes:

Reply:
:heart_decoration:

Reply:
Shouldn't we use `HEADER_SIZE` here too?

```suggestion
            let mut header_vec = Cursor::new(&src[0..HEADER_SIZE]);
```